### PR TITLE
Add documentation to several functions in jax.lax.linalg.

### PR DESCRIPTION
**Pull request by hawkinsp**

### CODE REVIEW

File: jax/_src/lax/linalg.py [hunk 1]
```diff
@@ -43,26 +43,133 @@
...
+      eigendecomposition by computing :math:`\\frac{1}{2}(x + x^H)`.
+
+  Returns:
+    A tuple ``(v, w)``.
+
+    ``v`` is an array with the same dtype as ``x`` (or its real counterpart if
+    complex) with shape ``[..., n]`` containing the eigenvalues of ``x``.
+
+    ``w`` is an array with the same dtype as ``x`` such that ``w[:, i]`` is the
+    eigenvector corresponding to ``v[i]``.
```
Comment (jakevdp) [inline jax/_src/lax/linalg.py:106]: should this be `w[..., :, i]` and `v[..., i]`?

File: jax/_src/lax/linalg.py [hunk 2]
```diff
@@ -43,26 +43,133 @@
...
+
+  Returns:
+    A pair of arrays ``(q, r)``.
+
+    Array ``q`` is a unitary (orthogonal) matrix,
+    with shape ``[..., m, m]`` if ``full_matrices=True``, or
+    ``[..., m, min(m, n)]`` if ``full_matrices=False``.
+
+    Array ``r`` is an upper-triangular matrix with shape ``[..., m, n]`` if
+    ``full_matrices=True``, or ``..., min(m, n), n]`` if
```
Comment (jakevdp) [inline jax/_src/lax/linalg.py:170]: missing `[` in ``[..., min(m, n), n]``

File: jax/_src/lax/linalg.py [hunk 3]
```diff
@@ -43,26 +43,133 @@
...
+  """
   if symmetrize_input:
     x = symmetrize(x)
   return jnp.tril(cholesky_p.bind(x))
 
 def eig(x, compute_left_eigenvectors=True, compute_right_eigenvectors=True):
+  """Eigendecomposition of a general matrix.
+
+  Nonsymmetric eigendecomposition is at present only implemented on CPU.
+  """
```
Comment (jakevdp) [inline jax/_src/lax/linalg.py:79]: Any reson not to document this more fully?

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: jax/_src/lax/linalg.py (hunk 3 above)
Comment (hawkinsp) [reply jax/_src/lax/linalg.py:79]: I ran out of time. We can improve it in a future PR.

**Review by jakevdp — APPROVED** [review]

**Pull request merged**